File: datadrop-api/src/main/java/com/web/datadropapi/Models/DirectoryDto.java
package com.web.datadropapi.Models;

import com.web.datadropapi.Enums.SharedState;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class DirectoryDto {
    private Long id;
    private String name;
    private LocalDate creationDate;
    private LocalDate lastModifiedDate;
    private SharedState sharedState;
    private List<FileDto> files;
    private List<SubdirectoryDto> subdirectories;
    private Long parentDirectoryId;
    private UserDto owner;
    private List<Long> sharedWithUsers;
}

File: datadrop-web/src/app/shared/models/DirectoryDto.ts
import { FileDto } from './FileDto';
import { UserDto } from './UserDto';

export interface DirectoryDto {
  id: number;
  name: string;
  creationDate: Date;
  lastModifiedDate: Date;
  sharedState: string;
  files: FileDto[];
  subdirectories: DirectoryDto[]; //null 2 layers deep
  parentDirectoryId: number;
  owner: UserDto;
  sharedWithUsers: number[];
}
